projects/gateway/pkg/translator/virtual_service_translator.go

```diff
@@ -330,10 +330,27 @@ func VirtualHostName(vs *v1.VirtualService) string {
 	return fmt.Sprintf("%v.%v", vs.GetMetadata().GetNamespace(), vs.GetMetadata().GetName())
 }
 
-// this function is written with the assumption that the routes will not be modified afterwards,
+// this function is written with the assumption that the routes will not be modified afterward,
 // and are in their final sorted form
-func validateRoutes(vs *v1.VirtualService, vh *gloov1.VirtualHost, reports reporter.ResourceReports) {
+func validateRoutesRegex(vs *v1.VirtualService, vh *gloov1.VirtualHost, reports reporter.ResourceReports) {
 	for _, rt := range vh.GetRoutes() {
+		options := rt.GetOptions()
+		if options != nil {
+			// validate HostRewrite regex
+			if options.GetHostRewritePathRegex() != nil {
+				_, err := regexp.Compile(options.GetHostRewritePathRegex().GetPattern().GetRegex())
+				if options.GetHostRewritePathRegex().GetPattern().GetRegex() != "" && err != nil {
```

Comment:
```suggestion
			if options.GetHostRewritePathRegex().GetPattern().GetRegex() != nil {
				_, err := regexp.Compile(options.GetHostRewritePathRegex().GetPattern().GetRegex())
				if err != nil {
```

It looks like this ought to work and would avoid unnecessarily calling `regexp.Compile()` when it has no chance of succeeding

Reply:
You can't call a nil check on a string so this would not work. I also think this might panic on trying to access a field of options.hostRewritePathRegex if it is nil